Automattic/wp-calypso
# Plans: update domain to plan nudge to use Banner component

This PR updates the domain to plan nudge to use a [Banner](https://wpcalypso.wordpress.com/devdocs/design/banner) component. h/t @Copons and @folletto 

See previous related PRs: #8635 and #9254

Other copy suggestions are welcome.

Desktop:
![screen shot 2017-01-17 at 10 52 07 am](https://cloud.githubusercontent.com/assets/1270189/22034964/069b5172-dca3-11e6-8a9e-c9ae64103bbe.png)

Mobile:
![screen shot 2017-01-17 at 10 52 19 am](https://cloud.githubusercontent.com/assets/1270189/22034968/0b615800-dca3-11e6-99f7-80b518657a1e.png)

### Testing Instructions:
- Navigate to http://calypso.localhost:3000/domains
- Select a wpcom site that has a paid domain, but a free plan
- We should see a nudge render below domains
- An impression analytics event should fire:
<img width="948" alt="screen shot 2017-01-16 at 3 59 00 pm" src="https://cloud.githubusercontent.com/assets/1270189/22003592/b2d25c8c-dc09-11e6-9ce2-e12261eb9a1c.png">
- Clicking on the nudge button takes us to checkout with a personal plan in cart
- The following analytics event should fire:
<img width="899" alt="screen shot 2017-01-16 at 3 59 23 pm" src="https://cloud.githubusercontent.com/assets/1270189/22003601/c0d7d2d0-dc09-11e6-99c1-eef50d9de930.png">

cc @rralian @sararosso @apeatling
